opendcs-rest-api/src/main/java/org/opendcs/odcsapi/dao/OpenDcsDatabaseFactory.java

```diff
@@ -16,49 +16,101 @@
 package org.opendcs.odcsapi.dao;
 
 import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
 import java.sql.SQLException;
+import java.sql.SQLRecoverableException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
 import javax.sql.DataSource;
 
-import decodes.cwms.CwmsDatabaseProvider;
 import decodes.db.DatabaseException;
-import decodes.sql.OracleSequenceKeyGenerator;
 import decodes.util.DecodesSettings;
-import opendcs.opentsdb.OpenTsdbProvider;
 import org.opendcs.database.DatabaseService;
 import org.opendcs.database.api.OpenDcsDatabase;
-import org.opendcs.odcsapi.hydrojson.DbInterface;
-import org.opendcs.spi.database.DatabaseProvider;
-import org.slf4j.Logger;
-import org.opendcs.utils.logging.OpenDcsLoggerFactory;
+import org.opendcs.odcsapi.dao.datasource.ConnectionPreparer;
+import org.opendcs.odcsapi.dao.datasource.ConnectionPreparingDataSource;
+import org.opendcs.odcsapi.dao.datasource.DelegatingConnectionPreparer;
+import org.opendcs.odcsapi.dao.datasource.DirectUserPreparer;
+import org.opendcs.odcsapi.dao.datasource.SessionOfficePreparer;
+import org.opendcs.odcsapi.dao.datasource.SessionTimeZonePreparer;
 
 public final class OpenDcsDatabaseFactory
 {
-	private static final Logger log = OpenDcsLoggerFactory.getLogger();
-	private static OpenDcsDatabase database;
+
+	private static final Map<String, OpenDcsDatabase> dbCache = new HashMap<>();
```

Comment:
I'd really like to avoid this if at all possible. At a minimum put some explanatory text as to why where's stuck maintaining multiple instances of the database class for future readers.